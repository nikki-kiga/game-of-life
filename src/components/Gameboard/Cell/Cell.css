button {
    outline: none;
}
.born {
    background: #6dd9c1;
} 

.alive {
    background: #ffea85;
}

.cell {
    padding: .5rem;
    cursor: pointer;
}

.cell:hover,
.cell:focus {
    background: #097287;
}

