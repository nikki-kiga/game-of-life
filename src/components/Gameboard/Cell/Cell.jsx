import React from 'react';
import PropTypes from 'prop-types';

import styles from './Cell.css';

function Cell({ alive, onClick, i, j }) {
  const aliveStyle = alive ? styles.alive : '';

  return (
    <button
      type='button'
      className = {`${styles.cell} ${aliveStyle}`}
      onClick={() => onClick(i, j)}>
    </button>
  );
}

Cell.propTypes = {
  alive : PropTypes.bool,
  onClick : PropTypes.func,
  i: PropTypes.number,
  j: PropTypes.number
};

export default Cell;

