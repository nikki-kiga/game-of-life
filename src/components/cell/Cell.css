button {
  outline: none;
  background: hsl(0, 0%, 10%);
}
.born {
  background: hsl(180, 60%, 50%);
}

.alive {
  background: hsl(180, 30%, 30%);
}

.cell {
  padding: 0.5rem;
  cursor: pointer;
  border-color: hsl(240, 50%, 37%);
}

.cell:hover,
.cell:focus {
  background: hsl(180, 80%, 50%);
}
