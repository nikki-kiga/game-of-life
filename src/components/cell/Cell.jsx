import React, { memo } from "react";
import PropTypes from "prop-types";

import styles from "./Cell.css";

/**
 * Cell component that returns buttons with specific styles
 * @component
 * @param {number} alive number representing cell life cycle
 * @param {func} onClick function executed on click
 * @param {number} i index of the row
 * @param {number} j index of column
 */
const Cell = memo(({ alive, onClick, i, j }) => {
  const aliveStyle = alive > 0 ? (alive > 1 ? styles.born : styles.alive) : "";

  return (
    <button
      type="button"
      className={`${styles.cell} ${aliveStyle}`}
      onClick={() => onClick(i, j)}
    ></button>
  );
});

Cell.propTypes = {
  alive: PropTypes.number,
  onClick: PropTypes.func,
  i: PropTypes.number,
  j: PropTypes.number,
};

export default Cell;
