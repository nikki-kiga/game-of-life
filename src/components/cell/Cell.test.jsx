import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Cell from './Cell';
import styles from './Cell.css';

describe('Cell component', () => {
  it('Cell renders correctly', () => {
    const tree = renderer.create(<Cell />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have born styling class', () => {
    const wrapper = shallow(<Cell alive={2} onClick={jest.fn()} i={1} j={1}/>);
    expect(wrapper.find(`.${styles.born}`)).toHaveLength(1);
  });
});
