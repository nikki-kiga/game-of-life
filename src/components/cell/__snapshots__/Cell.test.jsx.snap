// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component Cell renders correctly 1`] = `
<button
  className="cell "
  onClick={[Function]}
  type="button"
/>
`;
